### PR TITLE
Switch to Github actions

### PR DESCRIPTION
**Pull request by ekohl**
This is a temporary target to the gha-test branch which has the skeleton to see if it works. It'll go out of draft once it's cleaned up.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup ruby
+        uses: actions/setup-ruby@v1
+        with:
+          ruby-version: 2.7
+      - uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
```
Comment (ekohl) [inline .github/workflows/main.yml:18]: I'm still not sure if I need to include the Ruby version or not if it's not a matrix. Thoughts?

Comment (wbclark) [reply .github/workflows/main.yml:18]: I think you could test this by:

1. Push a commit with empty `.github/workflows/main.yml` directly to master in order to get CI running
2. Push this branch with a 2nd commit that has some dummy change that would be reflected in `bundle exec msync update --noop`
3. When satisfied with how CI is running, remove the 2nd commit and force push

**Review by ehelms — APPROVED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (ekohl) [reply .github/workflows/main.yml:18]: That's not really the point. It's more that the cache probably isn't invalidated if you later upgrade to Ruby 3.0.

Comment (ekohl) [reply .github/workflows/main.yml]: > Push a commit with empty `.github/workflows/main.yml` directly to master in order to get CI running

Also, in this PR I had a better workaround that appears to work. First create a branch (I created `gha-test`) which includes the skeleton. That triggers a build. Then you switch the base branch to master without the skeleton. That keeps the git log clean while it still tests.

**Pull request merged**